generator/osm_source.cpp

```diff
@@ -524,6 +532,30 @@ class MainFeaturesEmitter : public EmitterBase
 };
 }  // anonymous namespace
 
+void SyncOfstream::Open(string const & fullPath)
+{
+  lock_guard<mutex> gard(m_mutex);
+  m_stream.open(fullPath, std::ofstream::out);
+}
+
+bool SyncOfstream::IsOpened()
+{
+  lock_guard<mutex> gard(m_mutex);
+  return m_stream.is_open() && !m_stream.fail();
+}
+
+void SyncOfstream::Write(uint32_t featureId, vector<osm::Id> const & osmIds)
+{
+  if (!IsOpened())
+    return;
+
+  lock_guard<mutex> gard(m_mutex);
+  m_stream << featureId << ", ";
+  for (osm::Id const & osmId : osmIds)
+    m_stream << osmId.OsmId() << ", ";
```

Comment:
I did it to make code easier. Anyway deserializer is ready if there's no comma at the end of line (ParseLineOfNumbers). Do you think it'll be better to add some lines of code and stop writing commas at the end?